src/cargo/sources/path.rs

```diff
@@ -1000,8 +1005,31 @@ fn read_nested_packages(
     //
     // TODO: filesystem/symlink implications?
     if !source_id.is_registry() {
-        for p in nested.iter() {
-            let path = paths::normalize_path(&path.join(p));
+        let workspace_root_cell: LazyCell<PathBuf> = LazyCell::new();
+
+        for (p, base) in nested.iter() {
+            let p = if let Some(base) = base {
+                let workspace_root = || {
+                    workspace_root_cell
+                        .try_borrow_with(|| {
+                            find_workspace_root(&manifest_path, gctx)?
+                                .ok_or_else(|| anyhow!("failed to find a workspace root"))
+                        })
+                        .map(|p| p.as_path())
+                };
+                // Pass in `None` for the `cargo-features` not to skip verification: when the
+                // package is loaded as a dependency, then it will be checked.
+                match lookup_path_base(base, gctx, &workspace_root, None) {
+                    Ok(base) => Cow::Owned(base.join(p)),
+                    Err(err) => {
+                        errors.push(err);
+                        continue;
+                    }
+                }
+            } else {
+                Cow::Borrowed(p)
+            };
+            let path = paths::normalize_path(&path.join(p.as_path()));
```

Comment:
Why is this needed?  Aren't we operating on a normalized package and don't we resolve path-bases during normaization?

For either of us to verify that is one of the reasons I recommend splitting commits into
- Test showing the existing behavior
- Fix/feature with test updates to show how the behavior changed